### PR TITLE
various fixes

### LINKED ISSUE #588: m64p isn't updating
m46p hasn't been updating https://github.com/m64p/m64p

Latest version is https://github.com/m64p/m64p/releases/tag/v2022.2.19
Scoop available version is: [m64p (2021.12.29)](https://github.com/m64p/m64p/releases/tag/v2022.2.3)

[The release after the current release](https://github.com/m64p/m64p/releases/tag/v2022.2.3) states "Changes in this release: Update URLs for main github repository" so maybe a URL is broken?

Manifest should probably make use of `github` for `checkver` instead of using custom `url`+`regex`:
- https://github.com/Calinou/scoop-games/blob/master/bucket/m64p.json#L43
- https://github.com/ScoopInstaller/scoop/wiki/App-Manifest-Autoupdate#special-cases

### PR DESCRIPTION
**Pull request by beyondmeat**
fixes #588

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<!-- Provide a general summary of your changes in the title above -->
+
+<!--
+  By opening this PR you confirm that will follow the contribution guidelines. You agree to submit a fully featured working manifest that creates shortcuts, bin shims, persists data, enable portable mode, and auto updates.
+-->
+
+[ ] I have read the [Contributing Guide](../CONTRIBUTING.md).
+
+## What emulator release type is this?
```
Comment (Calinou) [inline .github/pull_request_template.md:9]: scoop-games contains more than just emulators :slightly_smiling_face: 

```suggestion
## What package release type is this?
```

It also contains launchers and various modding tools, so "game" is too specific as well.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] autoupdate and checkver entry [](https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate)
+- [ ] [persist](https://github.com/ScoopInstaller/Scoop/wiki/Persistent-data) defined with config/data/user/portable/textures/saves folder(s) specific for the app?
+- [ ] a [pre_install](https://github.com/ScoopInstaller/Scoop/wiki/Pre--and-Post-install-scripts) script to auto-enable portable mode (if needed)?
+  - [ ] script pre-creates generated files listed in persist? (solves linking issue with generated files that need to be persisted)
+- [ ] license identifier and url
+- [ ] a short terse description matching existing style.
+- [ ] url to the release along with its sha256 hash
+- [ ] bin entries
+    - [ ] if beta, dev, nightly, canary, etc: does bin have rewrite with variant appended to the name?
+- [ ] I tested the emulator and it runs in portable mode. I have verified the symlinks are correct in the persists folder.
```
Comment (Calinou) [inline .github/pull_request_template.md:25]:
```suggestion
- [ ] I tested the package and it runs in portable mode. I have verified the symlinks are correct in the persists folder.
```

**Comment by beyondmeat** [comment]
g2g

File: bucket/mgba-dev.json [hunk 3]
```diff
@@ -3,34 +3,43 @@
...
+            "url": "https://s3.amazonaws.com/mgba/build/mGBA-build-2022-03-08-win64-7296-54bccfd2cdd128fc35cc73fcd5aab8f4c0334b38.7z",
+            "hash": "fc07e1615a65956f0b73e79b5b62265eac2f67e3f4afd2a213a6b32f7e2b740f",
+            "extract_dir": "mGBA-build-2022-03-08-win64-7296-54bccfd2cdd128fc35cc73fcd5aab8f4c0334b38"
+        }
     },
+    "pre_install": [
+        "if (!(Test-Path \"$persist_dir\\qt.ini\")) {",
+        "   New-Item \"$dir\\qt.ini\" -Type File | Out-Null",
+        "}",
+        "if (!(Test-Path \"$persist_dir\\config.ini\")) {",
```
Comment (BlackPowerade) [inline bucket/mgba-dev.json:25]: Why is the config population left out? Without it many of the persisted directories are unused.

File: bucket/mgba.json [hunk 4]
```diff
@@ -6,18 +6,14 @@
...
-            "   'cheats', 'patch', 'savegame', 'savestate', 'screenshot' | ForEach-Object {",
-            "       Add-Content -Path \"$dir\\config.ini\" -Value $_`Path`=$persist_dir\\$_ }",
-            "}"
-        ]
-    },
+    "pre_install": [
+        "if (!(Test-Path \"$persist_dir\\qt.ini\")) {",
+        "   New-Item \"$dir\\qt.ini\" -Type File | Out-Null",
+        "}",
+        "if (!(Test-Path \"$persist_dir\\config.ini\")) {",
```
Comment (BlackPowerade) [inline bucket/mgba.json:13]: Same here, why leave out the config population?

File: bucket/mgba-dev.json (hunk 3 above)
Comment (beyondmeat) [reply bucket/mgba-dev.json:25]: mgba is now portable by default.

Comment (BlackPowerade) [reply bucket/mgba-dev.json:25]: Ah I see. Nevermind then.

**Pull request merged**